### DISCUSSION TITLE
Pickle vs. JSON

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvista#3263] [Q&A]
Has anyone given any thought as to whether we should switch from saving our datasets using `pickle` to using `JSON` or perhaps use a [safer unpickle](https://github.com/CensoredUsername/picklemagic)?

**Comment by MatthewFlamm** [comment]
I prefer to store my datasets in vtk xml files.

**Reply by banesullivan** [reply]
Yeah, I think we need to stick with an established data format, best if maintained by and for VTK, rather than implementing anything of our own.

It's nice that with pickle we can just save the contents of the VTK XML file in the "pickle", but we likely cannot with JSON (at least not in a more efficient way than what's done with pickle already).

**Reply by adam-grant-hendry** [reply]
Ya, XML seems more appropriate since you have namespaces and can make custom types, whereas with JSON you only have standard types. It's a larger file size, but probably better at explaining VTK data than JSON could.
